pylink/jlink.py

```diff
@@ -927,7 +928,7 @@ def connect(self, chip_name, speed='auto', verbose=False):
         # Required in the event that the device is unsecured, as it will be
         # mass erased here if the system is a Windows system and the user has
         # opted to erase the device.
-        time.sleep(2)
+        time.sleep(wait_after_connect_sec)
```

Comment:
I suspect adding a load bearing delay here isn't the right solution. @hkpeprah do we have a method of polling the host or target for when this operation might complete? It seems like that might be a cleaner solution than forcing callers to pull a number out of their rear for every target they support.

Reply:
Added unit test

Reply:
Yah, the load bearing sleep probably isn't the right thing to do here.  We probably want to remove it altogether.  I don't believe we need to poll the host/target, just issue any no-op command that would block.

Reply:
Is the current command after this suitable?  self._dll.JLINKARM_DEVICE_GetInfo(-1, 0)

Reply:
Unfortunately not.  That information is internal, so it doesn't talk to the target to get it.

Reply:
@hkpeprah What would be the ideal command?

Reply:
Hm.  Maybe querying if the CPU is halted via `.halted()`?  Would have to wrap it in a try-catch, but that should block.  We wouldn't/shouldn't have to care about the return value.

Reply:
PTAL, thanks